gdx/src/com/badlogic/gdx/graphics/g2d/ArrayTextureSpriteBatch.java

```diff
@@ -0,0 +1,1607 @@
+	 * the ones expected for shaders set with {@link #setShader(ShaderProgram)}.
+	 * <p>
+	 * <b>Remember: VRAM usage will be roughly maxTextureWidth * maxTextureHeight * maxConcurrentTextures * 4 byte plus some
+	 * overhead!</b>
+	 * 
+	 * @param maxSprites The maximum number of sprites in a single batched draw call. Upper limit of 8191.
+	 * @param maxTextureWidth Set as wide as your widest texture.
+	 * @param maxTextureHeight Set as tall as your tallest texture.
+	 * @param maxConcurrentTextures Set to the maximum number of textures you want to use ideally, grossly oversized values waste
+	 *           VRAM.
+	 * @param defaultShader The default shader to use. This is not owned by the ArrayTextureSpriteBatch and must be disposed
+	 *           separately. Remember to incorporate the fragment-/vertex-shader changes required for the use of an array texture
+	 *           as demonstrated by the default shader.
+	 * @throws IllegalStateException Thrown if the device does not support GLES 3.0 and by extension: GL_TEXTURE_2D_ARRAY and
+	 *            Framebuffer Objects. Make sure to implement a Fallback to {@link SpriteBatch} in case Texture Arrays are not
+	 *            supported on a device. */
+	public ArrayTextureSpriteBatch (int maxSprites, int maxTextureWidth, int maxTextureHeight, int maxConcurrentTextures,
+		ShaderProgram defaultShader) throws IllegalStateException {
+
+		if (Gdx.gl30 == null) {
+			throw new IllegalStateException("GL30 is not available. Remember to set \"useGL30 = true\" in your application config.");
+		}
+
+		// 32767 is max vertex index, so 32767 / 4 vertices per sprite = 8191 sprites max.
+		if (maxSprites > 8191) {
+			throw new IllegalArgumentException("Can't have more than 8191 sprites per batch: " + maxSprites);
+		}
+
+		if (maxConcurrentTextures < 1 || maxConcurrentTextures > 256) {
+			throw new IllegalArgumentException("maxConcurrentTextures out of range [1,256]: " + maxConcurrentTextures);
+		}
+
+		if (maxTextureWidth < 1 || maxTextureHeight < 1) {
+			throw new IllegalArgumentException(
+				"Maximum Texture width / height must both be greater than zero: " + maxTextureWidth + " / " + maxTextureHeight);
+		}
+
+		maxTextureLevels = maxConcurrentTextures;
+
+		this.maxTextureWidth = maxTextureWidth;
+		this.maxTextureHeight = maxTextureHeight;
+
+		invMaxTextureWidth = 1f / maxTextureWidth;
+		invMaxTextureHeight = 1f / maxTextureHeight;
+
+		if (defaultShader == null) {
+			shader = createDefaultShader();
+			ownsShader = true;
+
+		} else {
+			shader = defaultShader;
+			ownsShader = false;
+		}
+
+		FBO_READ_INTBUFF = ByteBuffer.allocateDirect(16 * Integer.BYTES).order(ByteOrder.nativeOrder()).asIntBuffer();
+
+		usedTextures = new Texture[maxTextureLevels];
+		usedTexturesLFU = new int[maxTextureLevels];
+		usedTexturesLFUPrevious = new int[maxTextureLevels];
+
+		arrayTextureMagFilter = GL30.GL_NEAREST;

```

Comment:
Hey, great work!
Just checked with my game-in-progess and I was wondering why it wlooked strange with your ArrayTextureSpriteBatch.
I normally use
arrayTextureMagFilter = GL30.GL_LINEAR;

It would be great if this was not hardcoded here but if it was a contructor parameter to make it more prominent what is going on here. I was very confused at first and didn't know what was going on...
I think some people will not get what's going on as other batches get this setting from totally other places, e.g. atlas settings (if I recall right...).

Reply:
It isn't hardcoded though 😁 there's already a chainable function just for that: `setArrayTextureFilter (int glTextureMagFilter, int glTextureMinFilter)`

Reply:
Yes, I saw that one, I just think it isn't prominent enough for people to notice 😉 and it runs the same code several times then.
I guess it's just a personal preference of mine.

Reply:
Alright, i'll add it 👌